content/blog/2022/2022-08-03-feature-flags-in-nodejs-backend-server-with-launchdarkly.md

```diff
@@ -0,0 +1,720 @@
+		// the default value is NOT a JSON STRING but, rather, AN ARRAY. In this case,
+		// we're defaulting to the empty array so the site can FAIL OPEN.
+		[]
+	);
+ 
+	// Check to see if the incoming request IP should be blocked.
+	if ( ipBlacklist.includes( request.ip ) ) { 
+		console.log( chalk.red( 'Blocking IP', chalk.bold( request.ip ) ) );
+		console.log( chalk.red.italic( `... blocking one of ${ ipBlacklist.length }`
+		+ ` known IP addresses.` ) ); 
+	} else { 
+		console.log( chalk.green( 'Allowing IP', chalk.bold( request.ip ) ) ); 
+	} 
+}
+```
+
+Next we can define the script as part of `package.json`:
+
+```json
+{
+	"scripts": {
+    	"blacklist": "nodemon blacklist_check.js"
+  	}
+}
+```
+
+Then we can execute the following command to run our app:
+
+```bash
+npm run blacklist
+```
+
+Finally when we run the above command it will print something like below:
+
+```bash
+info: [LaunchDarkly] Initializing stream processor to receive feature flag updates
+info: [LaunchDarkly] Opened LaunchDarkly stream connection
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+```
+
+Now we can go to the UI, add this IP and redeploy the variation. We will immediately see dynamically the app has picked the blocked IP and start printing it out something like below:
+
+```bash
+info: [LaunchDarkly] Initializing stream processor to receive feature flag updates
+info: [LaunchDarkly] Opened LaunchDarkly stream connection
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+```
+
+So this helped us to minimistically add an administrative feature to our existing microservice.
+
+### Update Batch Size while Performing Batch Operations on the Fly
+
+Consider that you have a batch task or job that needs to be executed. But we might need to update or somehow ramp-up the number of operations as per the need. So, if we think about it, we might need to first run this job behind some kind of scheduler and then we might need to pass some environment variables. Now if we want to update that number we always would need to restart our application.
+
+But we can just make use of our multivariate options in LaunchDarkly to define a batch size. We can dynamically change this values so that it reflects in the code accordingly.
+
+{{% image alt="Batch Multivariate option" src="images/posts/nodejs-backend-launchdarkly/Batch_Size_Count.png" %}}
```

Comment:
Again, please focus on the variations section of the screen.

Reply:
I have replaced this section with "Schedule Cron Jobs Dynamically" which is also an interesting topic specific to admin operations similar like "Managing Batch Job Size". Please review and let me know if this is fine.